teamscale-gradle-plugin/src/main/kotlin/com/teamscale/config/JUnitReportConfiguration.kt

```diff
@@ -2,12 +2,26 @@ package com.teamscale.config
 
 import com.teamscale.client.EReportFormat
 import org.gradle.api.Project
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.FileCollection
 import org.gradle.api.tasks.testing.Test
+import java.io.File
 
 /** Configuration for the JUnit results report. */
 open class JUnitReportConfiguration(project: Project, task: Test) :
     ReportConfigurationBase(EReportFormat.JUNIT, project, task) {
-    init {
-        destination.set(task.reports.junitXml.outputLocation)
+
+    /** The destination where the report should be written to/read from. */
+    var destination: DirectoryProperty = task.reports.junitXml.outputLocation
+
+    /** Allows to set the destination as a string, which is resolved as File internally. */
+    fun setDestination(destination: String) {
+        this.destination.set(project.objects.directoryProperty().fileValue(File(destination)))
+    }
```

Comment:
Not sure if this is done before, but if not, shouldn't we somehow handle wrong destinations here? Like destination is a directory, not readable/writable?

Reply:
The destination will likely not exist at that point in time as it will be created by another task after the configuration phase. For the tasks that produce it Gradle will internally check that the folder is writeable. Whether the file actually exists and is readable is determined in the actual task.